Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py

```diff
@@ -2448,34 +2448,59 @@ def get_remote_data_command(client: Client, args: Dict[str, Any], params: Dict)
 
     # Handle closing ticket/incident in XSOAR
     close_incident = params.get('close_incident')
-    if ticket.get('closed_at') and close_incident == 'closed' \
-            or ticket.get('resolved_at') and close_incident == 'resolved':
-        demisto.debug(f'ticket is closed: {ticket}')
-        entries.append({
-            'Type': EntryType.NOTE,
-            'Contents': {
-                'dbotIncidentClose': True,
-                'closeNotes': f'From ServiceNow: {ticket.get("close_notes")}',
-                'closeReason': converts_state_close_reason(ticket.get("state"))
-            },
-            'ContentsFormat': EntryFormat.JSON
-        })
+    if close_incident != 'None':
+        server_close_custom_state = params.get('server_close_custom_state')
+
+        if server_close_custom_state or (ticket.get('closed_at') and close_incident == 'closed') \
+                or (ticket.get('resolved_at') and close_incident == 'resolved'):
+            demisto.debug(f'ticket is closed: {ticket}')
+            entries.append({
+                'Type': EntryType.NOTE,
+                'Contents': {
+                    'dbotIncidentClose': True,
+                    'closeNotes': f'From ServiceNow: {ticket.get("close_notes")}',
+                    'closeReason': converts_state_close_reason(ticket.get("state"), server_close_custom_state)
+                },
+                'ContentsFormat': EntryFormat.JSON
+            })
 
     demisto.debug(f'Pull result is {ticket}')
     return [ticket] + entries
 
 
-def converts_state_close_reason(ticket_state: Optional[str]):
+def converts_state_close_reason(ticket_state: Optional[str], server_close_custom_state: Optional[str]):
     """
-    determine the XSOAR closeReason based on the Service Now ticket state.
+    determine the XSOAR incident close reason based on the Service Now ticket state.
+    if 'Mirrored XSOAR Ticket custom close state code' parameter is set, the function will try to use it to
+    determine the close reason (should be corresponding to a user-defined list of close reasons in the server configuration).
+    then it will try using 'closed' or 'resolved' state, if set using 'Mirrored XSOAR Ticket closure method' parameter.
+    otherwise, it will use the default 'out of the box' server incident close reason.
     Args:
         ticket_state: Service now ticket state
+        server_close_custom_state: server close custom state parameter
     Returns:
         The XSOAR state
     """
-    if ticket_state in ['6', '7']:
+
+    # check if server_close_custom_state parameter is set, and if so, try to parse it, and use it to return a custom close reason
```

Comment:
Lets comment to get better understanding of this part :)

Reply:
Done,